analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -49,27 +49,17 @@ public abstract class MethodsShouldNotHaveIdenticalImplementationsBase<TSyntaxKi
                         return;
                     }
 
-                    var methods = GetMethodDeclarations(c.Node).ToList();
+                    var methodsToHandle = new LinkedList<TMethodDeclarationSyntax>(GetMethodDeclarations(c.Node));
 
-                    var alreadyHandledMethods = new HashSet<TMethodDeclarationSyntax>();
-
-                    foreach (var method in methods)
+                    while (methodsToHandle.Count > 0)
                     {
-                        if (alreadyHandledMethods.Contains(method))
-                        {
-                            continue;
-                        }
-
-                        alreadyHandledMethods.Add(method);
-
-                        var duplicates = methods.Except(alreadyHandledMethods)
-                                                .Where(m => AreDuplicates(method, m))
-                                                .ToList();
-
-                        alreadyHandledMethods.UnionWith(duplicates);
+                        var method = methodsToHandle.First.Value;
+                        methodsToHandle.RemoveFirst();
+                        var duplicates = methodsToHandle.Where(x => AreDuplicates(method, x)).ToList();
 
                         foreach (var duplicate in duplicates)
                         {
+                            methodsToHandle.Remove(duplicate);
```

Comment:
This sounds like a task for `HashSet` instead of `LinkedList`

Reply:
see discussion here https://github.com/SonarSource/sonar-dotnet/pull/5623#discussion_r864700090